boa_macros/src/lib.rs

```diff
@@ -57,11 +57,59 @@
     clippy::nursery,
 )]
 
+use std::collections::HashSet;
+
 use proc_macro::TokenStream;
 use quote::quote;
-use syn::{parse_macro_input, LitStr};
+use syn::{parse::Parse, parse_macro_input, punctuated::Punctuated, Error, LitStr, Token};
 use synstructure::{decl_derive, AddBounds, Structure};
 
+struct Literals(Vec<LitStr>);
+
+impl Parse for Literals {
+    fn parse(input: syn::parse::ParseStream<'_>) -> syn::Result<Self> {
+        let parsed = Punctuated::<LitStr, Token![,]>::parse_terminated(input)?;
+        let strings = parsed.into_iter().collect::<Vec<_>>();
+        check_duplicates(&strings)?;
+        Ok(Self(strings))
+    }
+}
+
+fn check_duplicates(entries: &[LitStr]) -> syn::Result<()> {
+    let mut keys = HashSet::new();
+    for entry in entries {
+        if !keys.insert(entry) {
+            return Err(Error::new_spanned(entry, "duplicate string"));
+        }
+    }
+    Ok(())
+}
+
+/// Create an ordered static array of `&[u16]` strings from a list of [`str`] literals.
+#[proc_macro]
+pub fn static_strings(input: TokenStream) -> TokenStream {
```

Comment:
Is this only used internally? If its only meant for internal use we can use the `#[doc(hidden)]` attribute

Reply:
Removed it because there were some perf problems with this approach. Reverted to a local thread map.